### ISSUE TITLE
converter-sample : compile errors

### ISSUE THREAD

**Issue by ovlnmurty** [mouse07410/asn1c#34]
Hi mouse07410,

i am using your latest buid https://github.com/mouse07410/asn1c/tree/vlm_master.
I compiled my asn1 files. it is successfully compiled and generated .c and .h for the asn structures.
when i try to build the converter-sample which is generated with make, it is giving error. below is the compiler output.

[root@localhost pcap]# make -f converter-example.mk
cc -I/usr/src/ovl/pcap3/asn1c-vlm_master/libasn1parser/ -I/usr/src/ovl/pcap3/asn1c-vlm_master/libasn1fix/ -I/usr/src/ovl/pcap3/asn1c-vlm_master/libasn1print/ -I/usr/src/ovl/pcap3/asn1c-vlm_master/libasn1compiler -I/usr/src/ovl/pcap3/asn1c-vlm_master/skeletons -DASN_PDU_COLLECTION -I. -o ProtocolIE-Container.o -c ProtocolIE-Container.c
In file included from Extension-GANSS-AlmanacModel.h:15:0,
                 from GANSS-AlmanacModel.h:16,
                 from GANSS-AlmanacAndSatelliteHealth.h:16,
                 from GANSSGenericAssistanceData.h:68,
                 from GANSS-GenericAssistanceDataList.h:44,
                 from ProtocolExtensionField.h:68,
                 from ProtocolExtensionContainer.h:2750,
                 from UC-ID.h:49,
                 from UTRAN-GANSSReferenceTimeResult.h:17,
                 from ProtocolIE-Field.h:19,
                 from ProtocolIE-Container.h:264,
                 from ProtocolIE-Container.c:8:
ProtocolIE-Single-Container.h:30:1: error: unknown type name ‘Extension_ReferenceTimeChoice_IE_t’
 typedef Extension_ReferenceTimeChoice_IE_t  ProtocolIE_Single_Container_104P0_t;
 ^
ProtocolIE-Single-Container.h:31:1: error: unknown type name ‘Extension_GANSS_AlmanacModel_IE_t’
 typedef Extension_GANSS_AlmanacModel_IE_t  ProtocolIE_Single_Container_104P1_t;
 ^
ProtocolIE-Single-Container.h:32:1: error: unknown type name ‘Extension_InformationExchangeObjectType_InfEx_RqstIE_t’
 typedef Extension_InformationExchangeObjectType_InfEx_RqstIE_t  ProtocolIE_Single_Container_104P2_t;
 ^
make: *** [ProtocolIE-Container.o] Error 1
[root@localhost pcap]#

i am attaching the asn1 files which i am compiling.

[PCAP-ASNs.zip](https://github.com/mouse07410/asn1c/files/1814242/PCAP-ASNs.zip)

Please let me know if you require any other info to fix it. thanks.

**Comment by brchiu** [comment]
@ovlnmurty, you can use `-pdu=all -fcompound-names -findirect-choice -fno-include-deps -gen-PER` as command line option to `asn1c` for code generation. If it still does not work, you can coalesce separate ASN.1 files into one ASN.1 file. I tried generate code from `pcap-14.0.0.asn1` and it compiled.

**Comment by mouse07410** [comment]
@brchiu where can I find `pcap-14.0.0.asn1` file, to try this myself? Thanks!

**Comment by mouse07410** [comment]
I tried compiling using the flags that @brchiu suggested - and everything compiled except for one problem: in some files one name was written as `DGPSCorrection`, while in others it was `DgpsCorrection`. This caused compiler error, which I remedied by manually renaming `DGPSCorrection` to `DgpsCorrection` (fewer entries to change).

After that everything compiled fine. But the test-data (that buffer of 50 bytes) fails to decode - see #33 

**Update** Reason it failed to decode was that it wasn't an `InitiateMessage` PDU - it was a `PCAP-PDU` PDU that *encapsulated* `InitiateMessage`. Once that was corrected, decoding worked OK.
